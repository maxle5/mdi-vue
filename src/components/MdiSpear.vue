
  <template>
      <svg :width="size" :height="size" :viewBox="viewBox">
          <path d="M16 9H16.41L3.41 22L2 20.59L15 7.59V9H16M16 4V8H20L22 2L16 4Z" style="fill: currentColor" />
      </svg>
  </template>

  <script lang="ts">
  import { defineComponent } from 'vue';

  export default defineComponent({
      name: "MdiSpear",
      props: {
          size: { 
              type: Number, 
              default: 24 
          }
      },
      computed: {
        viewBox() {
          return `0 0 ${this.size} ${this.size}`;
        }
      }
  })
  </script>
