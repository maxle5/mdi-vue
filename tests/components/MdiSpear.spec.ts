
import { mount } from "@vue/test-utils";
import { expect, test } from "vitest";
import MdiSpear from "../../src/components/MdiSpear.vue";

test("MdiSpear snapshot", () => {
  const wrapper = mount(MdiSpear, {});
  expect(wrapper.html()).toMatchSnapshot();
});
