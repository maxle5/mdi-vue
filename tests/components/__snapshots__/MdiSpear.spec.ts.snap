// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MdiSpear snapshot 1`] = `
"<svg width="24" height="24" viewBox="0 0 24 24">
  <path d="M16 9H16.41L3.41 22L2 20.59L15 7.59V9H16M16 4V8H20L22 2L16 4Z" style="fill: currentColor;"></path>
</svg>"
`;
